CMakeLists.txt

```diff
@@ -14,7 +13,16 @@ enable_testing()
 include(SetupGTest)
 
 # C++ version
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+if ("${CMAKE_VERSION}" LESS "3.1")
```

Comment:
Supporting 14.04 is a good idea for atleast another year. You are allowed to require newer versions of CMake if you want to. 
I would like to point out that, while I love the new CMake standard flags as much as anyone, you have managed to turn one "not as pretty as it could be" line into 10 "more complicated than it needs to be" lines.
I'd recommend either just using the old style cxx_flags version or setting your minimum cmake version higher.